spec/models/invoice_item_spec.rb

```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+RSpec.describe InvoiceItem do 
+  describe "relationships" do 
+    it { should belong_to :invoice }
+    it { should belong_to :item }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :quantity }
+    it { should validate_presence_of :unit_price }
+    it { should validate_presence_of :status }
```

Comment:
Same for here with the enums

I think enums have a more specific shouldamatcher than can be used for validations, allows checking of values too.
something like

it { should define_enum_for(:status).with_values([pending: 0, packaged: 1, shipped: 2]) }

Reply:
completed, no `[]`